tests/grids/test_grid_extrusion.py

```diff
@@ -0,0 +1,686 @@
+
+        assert g_3d.num_nodes == 10
+        assert g_3d.num_cells == 2
+        assert g_3d.num_faces == 10
+
+        nodes = np.tile(nodes_2d, nz)
+        nodes[2, :5] = z[0]
+        nodes[2, 5:] = z[1]
+        assert np.allclose(g_3d.nodes, nodes)
+
+        fn_ind = np.array(
+            [
+                0,
+                1,
+                6,
+                5,
+                1,
+                2,
+                7,
+                6,
+                2,
+                3,
+                8,
+                7,
+                3,
+                4,
+                9,
+                8,
+                4,
+                0,
+                5,
+                9,
+                1,
+                4,
+                9,
+                6,
+                0,
+                1,
+                4,
+                1,
+                2,
+                3,
+                4,
+                5,
+                6,
+                9,
+                6,
+                7,
+                8,
+                9,
+            ]
+        )
+        fn_ind_ptr = np.array([0, 4, 8, 12, 16, 20, 24, 27, 31, 34, fn_ind.size])
+
+        fn_3d = sps.csc_matrix(
+            (np.ones(fn_ind.size), fn_ind, fn_ind_ptr), shape=(10, 10)
+        )
+        assert np.allclose(g_3d.face_nodes.toarray(), fn_3d.toarray())
+
+        cf_3d = np.array(
+            [
+                [1, 0],
+                [0, -1],
+                [0, 1],
+                [0, 1],
+                [-1, 0],
+                [1, -1],
+                [-1, 0],
+                [0, -1],
+                [1, 0],
+                [0, 1],
+            ]
+        )
+        assert np.allclose(g_3d.cell_faces.toarray(), cf_3d)
+
+        assert check_cell_map(cell_map, g_2d, nz - 1)
+        assert check_face_map(face_map, g_2d, nz - 1)
+
+
+@pytest.mark.parametrize(
+    "g,target_dim",
+    [
+        (pp.PointGrid(np.zeros((3, 1))), 1),
+        (pp.CartGrid(1), 2),
+        (pp.CartGrid([1, 1]), 3),
+    ],
+)
+def test_dimension_of_extruded_grid(g, target_dim):
+    z = np.arange(2)
+    gn, *_ = pp.grid_extrusion.extrude_grid(g, z)
+    assert gn.dim == target_dim
+
+
+def test_3D_grid_cannot_be_extruded():
+    g = pp.CartGrid([1, 1, 1])
+    z = np.arange(2)
+    with pytest.raises(ValueError):
+        pp.grid_extrusion.extrude_grid(g, z)
+
+
+def test_extrusion_with_single_fracture():
+    f = np.array([[1, 3], [1, 1]])
+    mdg = pp.meshing.cart_grid([f], [4, 2])
+
+    sd_2 = mdg.subdomains(dim=2)[0]
+    sd_1 = mdg.subdomains(dim=1)[0]
+
+    z = np.array([0, 1, 2])
+    mdg_new, _ = pp.grid_extrusion.extrude_grid_bucket(mdg, z)
+
+    for dim in range(mdg.dim_max()):
+        assert len(mdg.subdomains(dim=dim)) == len(mdg_new.subdomains(dim=dim + 1))
+
+    # Get hold of interface grids. We know there is a single interface in each
+    # mixed-dimensional grid, so make a loop and abort it.
+    for intf in mdg.interfaces():
```

Comment:
It looks like this test was in its original form incomplete, or more was intended at least